python/lib/mesh.cc

```diff
@@ -2,126 +2,315 @@
-                     "Set boundary ID's using a logical volume.");
-  mesh_continuum.def("SetupOrthogonalBoundaries", &MeshContinuum::SetupOrthogonalBoundaries, "...");
+    Parameters
+    ----------
+    log_vol: pyopensn.logvol.LogicalVolume
+        ???
+    mat_id: int
+        ???
+    sense: bool
+        ???
+    )",
+    py::arg("log_vol"),
+    py::arg("mat_id"),
+    py::arg("sense")
+  );
+  mesh_continuum.def(
+    "SetBoundaryIDFromLogicalVolume",
+    &MeshContinuum::SetBoundaryIDFromLogical,
+    R"(
+    Set boundary ID's using a logical volume.
 
+    Parameters
+    ----------
+
+    log_vol: pyopensn.logvol.LogicalVolume
+        ???
+    boundary_name: str
+        ???
+    sense: bool
+        ???
+    )",
+    py::arg("log_vol"),
+    py::arg("boundary_name"),
+    py::arg("sense") = true
+  );
+  mesh_continuum.def(
+    "SetupOrthogonalBoundaries",
+    &MeshContinuum::SetupOrthogonalBoundaries,
+    "???"
+  );
+  mesh_continuum.def(
+    "ExportToPVTU",
+    [](std::shared_ptr<MeshContinuum> self, const std::string& file_name) {
+        MeshIO::ToPVTU(self, file_name);
+    },
+    R"(
+    Write grid cells into PVTU format.
+
+    Parameters
+    ----------
+    file_base_name: str
+        Base name of the output file.
+    )",
+    py::arg("file_base_name")
+  );
+
+  // Surface mesh
   auto surface_mesh = py::class_<SurfaceMesh, std::shared_ptr<SurfaceMesh>>(
-    mesh, "SurfaceMesh",
+    mesh,
+    "SurfaceMesh",
     R"(
-      Surface mesh.
+    Surface mesh.
 
-      Wrapper of :cpp:class:`opensn::SurfaceMesh`.
-    )");
+    Wrapper of :cpp:class:`opensn::SurfaceMesh`.
+    )"
+  );
+  surface_mesh.def(
+    py::init(
+      [](py::kwargs& params)
+      {
+        return SurfaceMesh::Create(kwargs_to_param_block(params));
+      }
+    ),
+    R"(
+    Construct a surface mesh.
+
+    This function does not take any arguments.
+    )"
+  );
+  surface_mesh.def(
+    "ImportFromOBJFile",
+    &SurfaceMesh::ImportFromOBJFile,
+    R"(
+    Loads a surface mesh from a wavefront .obj file.
+
+    Parameters
+    ----------
+    file_name: str
+        ???
+    as_poly: bool, default=False
+        ???
+    transform: pyopensn.math.Vector3, default=(0.0, 0.0, 0.0)
+        ???
+    )",
+    py::arg("file_name"),
+    py::arg("as_poly") = false,
+    py::arg("transform") = Vector3()
+  );
+  surface_mesh.def(
+    "ImportFromTriangleFiles",
+    &SurfaceMesh::ImportFromTriangleFiles,
+    R"(
+    Loads a surface mesh from triangle's file format.
+
+    Parameters
+    ----------
+    file_name: str
+        ???
+    as_poly: bool
+        ???
+    )",
+    py::arg("file_name"),
+    py::arg("as_poly")
+  );
+  surface_mesh.def(
+    "ImportFromMshFiles",
+    &SurfaceMesh::ImportFromMshFiles,
+    R"(
+    Loads a surface mesh from gmsh's file format.
+
+    Parameters
+    ----------
+    file_name: str
+        ???
+    as_poly: bool
+        ???
+    )",
```

Comment:
str: filename mesh for the surface mesh
as_poly: if the surface mesh is allowed to contain polygonal facets (as opposed to only triangular faces)
transform: mesh transformation (I think if should be changed to translation as this is the only transformation that is allowed)

create an issue for all these surface mesh stuff and assign it to me for further review